### PR TITLE
Add distro version checking in ubuntu-setup

### PR DESCRIPTION
**Pull request by jgkamat**
Adds linux distro checking to ubuntu-setup. Still needed to be tested on ubuntu 15.04 but I think it works. I'll test that later this week.

### CODE REVIEW

File: util/ubuntu-setup [hunk 1]
```diff
@@ -12,17 +12,33 @@ HELP_STR="Usage: ./ubuntu-setup [-y | --yes] [-n | --noclobber] [-c | --clobber]
...
+if cat /etc/os-release | grep -iq '^NAME=.*Debian'; then # dont add repositories on debian, compatability issues?
+    echo "[WARN] You are using a flavor of debian. This configuration is not officially supported..." >&2
     ADD_REPOS=false
     FIRMWARE=false
+elif cat /etc/os-release | grep -iq '^NAME=.*Ubuntu'; then # We are using a version of ubuntu...
+    if cat /etc/os-release | grep -iq '^VERSION=.*14.04'; then # Using Ubuntu 14.04
+        echo "Ubuntu 14.04 Detected..."
+        ADD_REPOS=true
+        FIRMWARE=true
+    elif cat /etc/os-release | grep -iq '^VERSION=.*15.04'; then # Using Ubuntu 14.04
```
Comment (justbuchanan) [inline util/ubuntu-setup]: *15.04 in comment

**Comment by justbuchanan** [comment]
Cool, I like it!  Want to test it out on 15.04 before merging?

**Comment by jgkamat** [comment]
Yup, I'm going to test it out on a 15.04 docker container tomorrow, I'll let you know how it goes!

**Comment by jgkamat** [comment]
LGTM, Version checking seems to work on stock ubuntu 15. No issues with compiling, everything seems good. This does not support upgrades of our stack from 14 -> 15, only clean installs.

**Comment by jgkamat** [comment]
@justbuchanan it would be nice to test this in a VM as well before merging, but this looks good right now in docker containers

**Comment by jgkamat** [comment]
@justbuchanan Bump. Any chance this could get merged before tomorrow?

**Comment by justbuchanan** [comment]
My bad, totally missed this.  Looks good to me!

**Pull request merged**